mesonbuild/coredata.py

```diff
@@ -576,7 +575,8 @@ def sanitize_dir_option_value(self, prefix: str, option: OptionKey, value: T.Any
             try:
                 value = value.relative_to(prefix)
             except ValueError:
-                raise MesonException(msg.format(option, value, prefix))
+                # Path is not relative, let’s return it as is.
+                pass
```

Comment:
We still need `msg` for the case where `..` is present. However, it should be reworded to be more specific, since now if `..` is present in an option, Meson errors out by telling you... that anything outside of the prefix is not permitted, when that is incorrect, you just need to do it without a very specific substring designed to cause trouble.

Reply:
I wonder if we should even care about `..` now that we relaxed the “everything is inside prefix”. It appears to originate from https://github.com/mesonbuild/meson/pull/6398 without much context. @scivision, do you remember if it was just to prevent escaping prefix or were there another purpose.

Reply:
I think going outside the prefix with ".." makes little sense. It should be either an absolute path or a path inside the prefix.

Reply:
I do not see an issue with `..` per-se – the produced denormalized paths might be somewhat unsightly but there are countless other ways to introduce de-normalization. Nevertheless, we do not really need `..` so I re-introduced the error message (with more specific wording).